utils/_context/_scenarios/endtoend.py

```diff
@@ -803,3 +822,90 @@ def get_junit_properties(self) -> dict[str, str]:
         result["dd_tags[systest.suite.context.appsec_rules_file]"] = self.weblog_container.appsec_rules_file or ""
 
         return result
+
+    def __eq__(self, other: object) -> bool:
+        """Compare EndToEndScenario instances excluding name, doc and weblog_env"""
+        if not isinstance(other, EndToEndScenario):
```

Comment:
Exclude inherited classes : 

```suggestion
        if type(other) is not EndToEndScenario:
```